lib/ex_unit/lib/ex_unit/server.ex

```diff
@@ -74,10 +81,20 @@ defmodule ExUnit.Server do
 
   # Called by the runner when --repeat-until-failure is used.
   def handle_call({:restore_modules, async_modules, sync_modules}, _from, state) do
+    {async_modules, async_groups} =
+      Enum.reduce(async_modules, {[], []}, fn
+        {nil, [module]}, {modules, groups} ->
+          {[{:module, module} | modules], groups}
+
+        {group, group_modules}, {modules, groups} ->
+          {[{:group, group} | modules], Map.put(groups, group, group_modules)}
+      end)
+
     {:reply, :ok,
      %{
        state
        | loaded: :done,
+         async_groups: async_groups,
          async_modules: :queue.from_list(async_modules),
```

Comment:
Should we reverse the async and sync modules here to restore them to the original order or does it not matter?

Reply:
Order matters because it makes it reproducible, as we push items to the queue at any time. But queue.to_list should preserve the order, as long as you insert them in the correct order. So using queues should be enough, no reverses.